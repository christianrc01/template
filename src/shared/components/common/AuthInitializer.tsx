import { useEffect, useState } from "react";
import { useMsal } from "@azure/msal-react";
import { InteractionStatus } from "@azure/msal-browser";
import { loginRequest } from "@/features/auth/controllers/authConfig";

function AuthInitializer() {
  const { instance, accounts, inProgress } = useMsal();
  const [initialized, setInitialized] = useState(false);

  useEffect(() => {
    if (inProgress === InteractionStatus.None) {
      setInitialized(true);
    }
  }, [inProgress]);

  useEffect(() => {
    if (initialized && accounts.length === 0) {
      console.log("🔐 No session. Redirecting to Azure login...");
      instance.loginRedirect(loginRequest);
    } else if (accounts.length > 0) {
      console.log("✅ Session active:", accounts[0].username);
    }
  }, [initialized, accounts, instance]);

  return null;
}

export default AuthInitializer;
