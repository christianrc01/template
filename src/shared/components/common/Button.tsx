import type { ButtonProps } from "@/shared/types/IShared";

function Button({
  variant = "primary",
  children,
  className = "",
  type = "button",
  ...props
}: ButtonProps) {
  const baseClasses = "px-4 py-2 rounded-full transition-colors font-medium";
  const variantClasses = {
    primary:
      "bg-blue-500 text-white hover:bg-blue-600 dark:bg-blue-600 dark:hover:bg-blue-700",
    secondary:
      "bg-gray-500 text-white hover:bg-gray-600 dark:bg-gray-600 dark:hover:bg-gray-700",
    danger:
      "bg-red-600 text-white hover:bg-red-700 focus:ring-red-500 dark:bg-red-700 dark:hover:bg-red-800 dark:focus:ring-red-600",
    "outline-white":
      "bg-gray-100 border border-white hover:bg-white hover:text-blue-600 dark:border-gray-500 dark:text-gray-600 dark:hover:bg-gray-700 dark:hover:text-white",
  };

  const variantStyle = variantClasses[variant] ?? variantClasses["primary"];

  return (
    <button
      type={type}
      className={`${baseClasses} ${variantStyle} ${className}`}
      {...props}
    >
      {children}
    </button>
  );
}

export default Button;
