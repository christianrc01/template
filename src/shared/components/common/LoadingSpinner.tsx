function LoadingSpinner() {
  return (
    <div
      className="grid place-items-center h-64"
      role="status"
      aria-label="Loading..."
    >
      <div
        data-testid="spinner"
        className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500"
        aria-hidden="true"
      ></div>
    </div>
  );
}

export default LoadingSpinner;
