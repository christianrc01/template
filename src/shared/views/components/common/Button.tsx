import type { CustomButtonProps } from "@/shared/types/IShared";
import { Button as KendoButton } from "@progress/kendo-react-buttons";

function Button({
  variant = "primary",
  children,
  className = "",
  type = "button",
  ...props
}: CustomButtonProps) {
  const variantClass = `k-button-${variant}`;

  return (
    <KendoButton
      type={type}
      className={`${variantClass} ${className}`}
      {...props}
    >
      <span className="whitespace-normal break-words text-center block">
        {children}
      </span>
    </KendoButton>
  );
}

export default Button;
