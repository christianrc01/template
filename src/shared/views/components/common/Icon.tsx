import clsx from "clsx";
import type { IconProps } from "@/shared/types/IShared";

function Icon({
  icon: IconComponent,
  className = "",
  stroke = "currentColor",
  variant = "success",
  ...props
}: IconProps) {
  const baseClass = "k-icon";
  const variantClass = `k-icon-${variant}`;

  return (
    <IconComponent
      className={clsx(baseClass, variantClass, className)}
      stroke={stroke}
      aria-hidden="true"
      {...props}
    />
  );
}

export default Icon;
