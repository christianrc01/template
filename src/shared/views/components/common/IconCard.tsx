import type { IconCardProps } from "@/shared/types/IShared";
import Text from "@/shared/views/components/common/Text";

function IconCard({ icon, name }: IconCardProps) {
  return (
    <div className="flex items-center gap-4 bg-[var(--color-bg)] rounded-lg p-4 shadow-sm">
      <div>{icon}</div>
      <Text
        as="span"
        className="text-[var(--color-text-primary)] text-sm break-all"
      >
        {name}
      </Text>
    </div>
  );
}

export default IconCard;
