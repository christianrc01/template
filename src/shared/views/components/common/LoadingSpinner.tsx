import { Loader } from "@progress/kendo-react-indicators";

function LoadingSpinner() {
  return (
    <div
      className="grid place-items-center h-64"
      role="status"
      aria-label="Loading..."
      data-testid="spinner"
    >
      <Loader
        size="large"
        type="infinite-spinner"
        themeColor="secondary"
        aria-hidden="true"
      />
    </div>
  );
}

export default LoadingSpinner;


