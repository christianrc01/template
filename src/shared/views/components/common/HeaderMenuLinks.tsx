import { Link } from "react-router-dom";
import Button from "@/shared/views/components/common/Button";
import type { HeaderMenuLinksProps } from "@/shared/types/IShared";
import { Fragment } from "react";

function HeaderMenuLinks({
  links,
  onLogout,
  onLinkClick,
}: HeaderMenuLinksProps) {
  return (
    <Fragment>
      {links.map((link) => (
        <Link
          key={link.to}
          to={link.to}
          onClick={onLinkClick}
          draggable={false}
        >
          {link.label}
        </Link>
      ))}
      <Button variant="primary" onClick={onLogout}>
        Logout
      </Button>
    </Fragment>
  );
}

export default HeaderMenuLinks;
