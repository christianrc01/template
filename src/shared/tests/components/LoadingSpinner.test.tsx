import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { LoadingSpinner } from "../../components/common/LoadingSpinner";

describe("LoadingSpinner Component", () => {
  it("renders with proper accessibility attributes", () => {
    render(<LoadingSpinner />);

    const spinnerContainer = screen.getByRole("status");
    expect(spinnerContainer).toBeInTheDocument();
    expect(spinnerContainer).toHaveAttribute("aria-label", "Loading...");

    const visualSpinner = screen.getByLabelText("Loading...").firstChild;
    expect(visualSpinner).toHaveAttribute("aria-hidden", "true");
  });

  it("has correct animation classes", () => {
    render(<LoadingSpinner />);
    const spinner = screen.getByLabelText("Loading...").firstChild;
    expect(spinner).toHaveClass("animate-spin");
  });

  it("has the correct container classes", () => {
    const { container } = render(<LoadingSpinner />);
    const spinnerContainer = container.firstChild;

    expect(spinnerContainer).toHaveClass("grid");
    expect(spinnerContainer).toHaveClass("place-items-center");
    expect(spinnerContainer).toHaveClass("h-64");
  });

  it("displays a spinner with proper animation and styling", () => {
    render(<LoadingSpinner />);
    const spinner = screen.getByRole("status").firstChild;

    expect(spinner).toHaveClass("animate-spin");
    expect(spinner).toHaveClass("rounded-full");
    expect(spinner).toHaveClass("h-12");
    expect(spinner).toHaveClass("w-12");
    expect(spinner).toHaveClass("border-t-2");
    expect(spinner).toHaveClass("border-b-2");
    expect(spinner).toHaveClass("border-blue-500");
  });

  it("has aria-role for accessibility", () => {
    render(<LoadingSpinner />);
    expect(screen.getByRole("status")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<LoadingSpinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
