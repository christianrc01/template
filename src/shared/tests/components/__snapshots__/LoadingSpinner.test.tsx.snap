// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading..."
    class="grid place-items-center h-64"
    role="status"
  >
    <div
      aria-hidden="true"
      class="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500"
    />
  </div>
</DocumentFragment>
`;
